File: Binary Tree Postorder Traversal/Binary Tree Postorder Traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
     bool comp(int i, int j){
        return (i > j);
    }
    
class Solution {
public:

    void recur(std::vector<int>& res, TreeNode* node){
        if(node == NULL)    return;

        
        recur(res, node->left);
                recur(res, node->right);
        res.push_back(node->val);
    }
    

    vector<int> postorderTraversal(TreeNode *root) {
        std::vector<int> res;
        res.clear();
        recur(res, root);
        //std::sort(res.begin(), res.end(), comp);
        return res;
    }
};

File: Binary Tree Postorder Traversal/Binary Tree Postorder Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        if(root == null){
            return new ArrayList<Integer>();
        }
        
        if(root.left != null && root.right != null){
            List<Integer> l1 = postorderTraversal(root.left);
            List<Integer> l2 = postorderTraversal(root.right);
            List<Integer> l3 = new ArrayList<Integer>();
            l3.add(root.val);
            l1.addAll(l2);
            l1.addAll(l3);
            return l1;
        }
        else if(root.left != null){
            List<Integer> l1 = postorderTraversal(root.left);
            List<Integer> l3 = new ArrayList<Integer>();
            l3.add(root.val);
            l1.addAll(l3);
            return l1;
        }
        else if(root.right != null){
            List<Integer> l1 = postorderTraversal(root.right);
            List<Integer> l3 = new ArrayList<Integer>();
            l3.add(root.val);
            l1.addAll(l3);
            return l1;
        }
        else{
            List<Integer> l1 = new ArrayList<Integer>();
            l1.add(root.val);
            return l1;
        }
    }
}
